share/eds/retriever.go

```diff
@@ -286,7 +286,10 @@ func (rs *retrievalSession) doRequest(ctx context.Context, q *quadrant) {
 				if rs.square.GetCell(uint(x), uint(y)) != nil {
 					return
 				}
-				rs.square.SetCell(uint(x), uint(y), share)
+				if err := rs.square.SetCell(uint(x), uint(y), share); err != nil {
+					// TODO: handle error
+					return
+				}
```

Comment:
As SetCell now internally checks if the Cell was written before, we can remove the GetCell call above(which does exactly this check). The error returned from SetCell can be safely ignored.